src/runtime_strings.js

```diff
@@ -191,12 +191,16 @@ function lengthBytesUTF8(str) {
   for (var i = 0; i < str.length; ++i) {
     // Gotcha: charCodeAt returns a 16-bit word that is a UTF-16 encoded code unit, not a Unicode code point of the character! So decode UTF16->UTF32->UTF8.
     // See http://unicode.org/faq/utf_bom.html#utf16-3
-    var u = str.charCodeAt(i); // possibly a lead surrogate
-    if (u >= 0xD800 && u <= 0xDFFF) u = 0x10000 + ((u & 0x3FF) << 10) | (str.charCodeAt(++i) & 0x3FF);
-    if (u <= 0x7F) ++len;
-    else if (u <= 0x7FF) len += 2;
-    else if (u <= 0xFFFF) len += 3;
-    else len += 4;
+    let c = str.charCodeAt(i); // possibly a lead surrogate
```

Comment:
Sadly we have not yet made the es6 jump to `let`/`const` in the generated code.  We should probably look into doing so ASAP.  The tracking bug for es6 output by default is https://github.com/emscripten-core/emscripten/issues/11984.